lib/jekyll/page.rb

```diff
@@ -184,4 +184,14 @@ def write?
       true
     end
   end
+
+  class PageWithoutAFile < Page
```

Comment:
I would prefer a different name such as `BasePage` or `BasicPage` here. The fact that it doesn't have a file is irrelevant, IMHO

Reply:
Naming has been a rough sea here at Jekyll AFAIR.. so I'll change it to whatever the majority zeros in..